fabfile.py

```diff
@@ -46,258 +68,276 @@
-    """ Set up the development environment.
-        If the dbuser and dbfile arguments are supplied, sets up
-        encrypted database storage and loads a DB dump. """
-
-    if dbuser is not None and dbfile is None:
-        raise Exception('You must provide a database dump file to load in the dbfile argument.')
-    if dbuser is None and dbfile is not None:
-        raise Exception('You must specify the PostgreSQL user that your database belongs to in the dbuser argument.')
-    using_db_dump = (dbuser is not None and dbfile is not None)
-
-    if using_db_dump:
-        load_db_dump(dbuser, dbfile)
-    else:
-        create_settings()
-        initialize_db()
-    setup_apache()
-    link_media()
+def gen_password(length):
+    return "".join([random.choice(string.letters + string.digits) for i in range(length)])
 
 @task
-def run_devserver():
-    """ Run Django dev server on port 8000. """
-    with esp_env():
-        sudo('python manage.py runserver 0.0.0.0:8000')
+def refresh():
+    """
+    Re-synchronize the remote environment with the codebase. For use when
+    switching branches or jumping around the history in git.
+
+      - upgrades and downgrades Python packages
+      - removes orphaned *.pyc files
+      - runs Django database migrations
+    """
+    ensure_environment()
+
+    with cd(env.rbase):
+        # Update the virtualenv with correct packages
+        run("pip install --upgrade -r " + env.rbase + "esp/requirements.txt")
+
+        # Clean up *.pyc files that contain stale code. In practice, the only
+        # files that cause problems are orphaned *.pyc's where the corresponding
+        # *.py file has been deleted (git leaves the *.pyc in place). Other than
+        # this, *.pyc files cannot be stale because git updates the lastmod time
+        # when changing a file, so a freshly checked out *.py file will always
+        # be newer than its *.pyc, triggering a recompile. This is an optimization
+        # over 'manage.py clean_pyc', which deletes all *pyc's in the source
+        # tree.. Note that this command is run in the Vagrant VM, since the host
+        # may not have find installed.
+        run("""find . -name '*.pyc' -exec bash -c 'test ! -f "${1%c}"' -- {} \; -delete""")
+
+        # Run Django migrations
+        manage("migrate")
+
+        # Recompile theme (run twice, to work around bug)
+        manage("recompile_theme")
+        manage("recompile_theme")
 
 @task
 def manage(cmd):
-    """ Run a manage.py command """
-    with esp_env():
-        sudo('python manage.py '+cmd)
-
-@task
-def update_deps():
-    sudo('%s/esp/update_deps.sh --virtualenv=%s' % (REMOTE_PROJECT_DIR, REMOTE_VIRTUALENV_DIR))
+    """
+    Run a manage.py command in the remote host.
+    """
+    ensure_environment()
+
+    if cmd in ["shell", "shell_plus"]:
+        # cd() doesn't work with open_shell
+        open_shell("(cd " + env.rbase + "esp && python manage.py " + cmd + "); exit")
```

Comment:
[open_shell](http://docs.fabfile.org/en/1.10/api/core/operations.html#fabric.operations.open_shell): it's supposed to provide a "better interactive experience" by not printing the hostname on every line, and doing some terminal-related stuff I'm unclear on.